src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java

```diff
@@ -13,35 +13,43 @@
  */
 public class DefaultClassLoaderProvider implements ClassLoaderProvider {
   public DefaultClassLoaderProvider() {
-    urls = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
+    customerUrls = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
+    workerUrls = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
   }
 
   /*
    * @see com.microsoft.azure.functions.reflect.ClassLoaderProvider#createClassLoader()
    */
   @Override
-  public ClassLoader createClassLoader() {
-    URL[] urlsForClassLoader = new URL[urls.size()];
-    urls.toArray(urlsForClassLoader);
+  public ClassLoader createClassLoader() throws IOException {
+    List<URL> urlsList = new ArrayList<>();
+    urlsList.addAll(this.customerUrls);
+    urlsList.addAll(this.workerUrls);
+    addUrlToSystemClassLoader(urlsList);
 
+    //TODO: validate below lines should not exist
```

Comment:
a little more context to this TODO might be helpful

Reply:
right! will add more context to this TODO item.